pkg/aws/throttle/config.go

```diff
@@ -0,0 +1,102 @@
+package throttle
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"golang.org/x/time/rate"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type throttleConfig struct {
+	operationPtn *regexp.Regexp
+	r            rate.Limit
+	burst        int
+}
+
+var _ pflag.Value = &ServiceOperationsThrottleConfig{}
+
+// serviceOperationsThrottleConfig is throttleConfig for each service's operations.
+// It supports to be configured using flags with format like "${serviceID}:${operationRegex}={rate}:{burst}"
+// e.g. "appmesh:DescribeMesh=1.3:5,appmesh:Create.*=1.7:3"
+// Note: default throttle for each service will be cleared if any override is set for that service.
+type ServiceOperationsThrottleConfig struct {
+	// service:operationRegex:config
+	value map[string][]throttleConfig
+}
+
+func (c *ServiceOperationsThrottleConfig) String() string {
+	if c == nil {
+		return ""
+	}
+
+	var configs []string
+	var serviceIDs []string
+	for serviceID := range c.value {
+		serviceIDs = append(serviceIDs, serviceID)
+	}
+	sort.Strings(serviceIDs)
+	for _, serviceID := range serviceIDs {
+		for _, operationsThrottleConfig := range c.value[serviceID] {
+			configs = append(configs, fmt.Sprintf("%s:%s=%v:%d",
+				serviceID,
+				operationsThrottleConfig.operationPtn.String(),
+				operationsThrottleConfig.r,
+				operationsThrottleConfig.burst,
+			))
+		}
+	}
+	return strings.Join(configs, ",")
+}
+
+func (c *ServiceOperationsThrottleConfig) Set(val string) error {
+	valueOverride := make(map[string][]throttleConfig)
+	configPairs := strings.Split(val, ",")
+	for _, pair := range configPairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) != 2 {
+			return errors.Errorf("%s must be formatted as serviceID:operationRegex=rate:burst", pair)
+		}
+		serviceIDOperationRegexPair := strings.Split(kv[0], ":")
+		if len(serviceIDOperationRegexPair) != 2 {
+			return errors.Errorf("%s must be formatted as serviceID:operationRegex", kv[0])
+		}
+		rateBurstPair := strings.Split(kv[1], ":")
```

Comment:
Would it be good to range limit the values users can choose for the rate/burst? Maybe good to prevent them from choosing unreasonable values on both ends (high and low)

Reply:
I don't think there are sensible high and low. (except they must be non-negative).
These are advanced configurations, and ppl should only override them if they got limit increase from aws side. (where they can got corresponding rate & burst config from aws)